### PR TITLE
[ruby] Add gemspec required_ruby_version setting.

### PR DESCRIPTION
**Pull request by cliffano**
Add required_ruby_version option to gemspec, defaulting to >= 1.9 considering 1.8 was retired a while ago.
Ruby petstore has been regenerated.

### CODE REVIEW

**Comment by wing328** [comment]
@cliffano thanks for the PR. Please rebase on the latest master to resolve the merge conflicts.

**Comment by cliffano** [comment]
@wing328 Conflict resolved.

File: modules/swagger-codegen/src/main/resources/ruby/gemspec.mustache [hunk 1]
```diff
@@ -17,6 +17,7 @@ Gem::Specification.new do |s|
   s.summary     = "{{gemSummary}}{{^gemSummary}}{{{appName}}} Ruby Gem{{/gemSummary}}"
   s.description = "{{gemDescription}}{{^gemDescription}}{{{appDescription}}}{{^appDescription}}{{{appName}}} Ruby Gem{{/appDescription}}{{/gemDescription}}"
   s.license     = "{{gemLicense}}{{^gemLicense}}{{{licenseInfo}}}{{^licenseInfo}}Apache 2.0{{/licenseInfo}}{{/gemLicense}}"
+  s.required_ruby_version = "{{gemRequiredRubyVersion}}{{^gemRequiredRubyVersion}}>= 1.9{{/gemRequiredRubyVersion}}"
```
Comment (wing328) [inline modules/swagger-codegen/src/main/resources/ruby/gemspec.mustache]: @cliffano similar to `gemLicense`, I think you will also need to add `gemRequiredRubyVersion` to RubyClientCodegen.java:

https://github.com/swagger-api/swagger-codegen/blob/e044f9d9bf50efb40beb7f623a8e21e8a134bc84/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/RubyClientCodegen.java#L32

Comment (cliffano) [inline modules/swagger-codegen/src/main/resources/ruby/gemspec.mustache]: @wing328 Thanks for the pointer, done in https://github.com/swagger-api/swagger-codegen/pull/3718/commits/b68b53975340f945d291f27fa75f70b4781aae8f following gemLicense pattern.

**Comment by wing328** [comment]
@cliffano PR merged into master. Thanks for the contribution.

**Pull request merged**